pkg/dfns-api-client/auth_transport.go

```diff
@@ -0,0 +1,276 @@
+package dfns_api_client
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"slices"
+	"strconv"
+
+	"github/dfns/dfns-sdk-go/pkg/credentials"
+)
+
+const (
+	UserActionHeader    = "x-dfns-useraction"
+	appIDHeader         = "x-dfns-appid"
+	nonceHeader         = "x-dfns-nonce"
+	authorizationHeader = "authorization"
+	contentTypeHeader   = "Content-Type"
+)
+
+type authTransport struct {
+	*DfnsApiOptions
+}
+
+func (auth *authTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	auth.setHeaders(req)
+
+	if ok := slices.Contains([]string{"POST", "PUT", "DELETE"}, req.Method); ok {
```

Comment:
nit: feels this check belongs to the `shouldPerformUserAction` method, can be done inside it ?